src/coreclr/jit/importer.cpp

```diff
@@ -9536,13 +9536,20 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             //
             methHnd = callInfo->hMethod;
         }
+        else if (call->AsCall()->IsDelegateInvoke() && opts.OptimizationEnabled() && !opts.IsOSR())
+        {
+            considerGuardedDevirtualization(call->AsCall(), rawILOffset, false, NO_METHOD_HANDLE, NO_CLASS_HANDLE,
+                                            nullptr);
+        }
```

Comment:
@AndyAyersMS I was seeing issues where we try to both do GDV and add probes in OSR variants. There are comments that say that we currently do not devirtualize in OSR variants here:
https://github.com/dotnet/runtime/blob/66bd8b193bdaa96e26308e6c056f0b27434428cd/src/coreclr/jit/importer.cpp#L20909-L20912

Do you know off the top of your head where we ensure that no GDV should happen for OSR variants?

Reply:
Ah, I think it is the early out of `impDevirtualizeCall` that I have mistakenly removed while extracting `impConsiderCallProbe`.

Reply:
Yeah, looks like you found it.